### PR TITLE
feat(sdk-crash-detection): Add empty feature

### PR DESCRIPTION
**Pull request by philippsuess**
This PR adds a feature flag disabled by default, the project id for the feature in the settings, and calls the placeholder class for the SDK crash detection (https://github.com/getsentry/sentry/issues/44342) in post processing.

This is the first PR of splitting up the POC for SDK crash detection (https://github.com/getsentry/sentry/pull/49928) into multiple PRs.

### CODE REVIEW

File: src/sentry/tasks/post_process.py [hunk 1]
```diff
@@ -1056,6 +1056,28 @@ def fire_error_processed(job: PostProcessJob):
...
+def sdk_crash_monitoring(job: PostProcessJob):
+    from sentry.utils.sdk_crashes.sdk_crash_detection import sdk_crash_detection
+
+    if job["is_reprocessed"]:
+        return
+
+    event = job["event"]
+
+    if not features.has("organizations:sdk-crash-detection", event.project.organization):
+        return
```
Comment (iker-barriocanal) [inline src/sentry/tasks/post_process.py:1068]: nit: should we move this to the top of the function?

File: src/sentry/tasks/post_process.py [hunk 2]
```diff
@@ -1056,6 +1056,28 @@ def fire_error_processed(job: PostProcessJob):
...
+
+    if settings.SDK_CRASH_DETECTION_PROJECT_ID is None:
+        logger.warning(
+            "SDK crash detection is enabled but SDK_CRASH_DETECTION_PROJECT_ID is not set."
+        )
+        return None
+
+    with metrics.timer("post_process.sdk_crash_monitoring.duration"):
+        with sentry_sdk.start_span(op="tasks.post_process_group.sdk_crash_monitoring"):
```
Comment (iker-barriocanal) [inline src/sentry/tasks/post_process.py:1077]: I believe you can start these two context managers at the same time:
```suggestion
    with metrics.timer("post_process.sdk_crash_monitoring.duration"), sentry_sdk.start_span(op="tasks.post_process_group.sdk_crash_monitoring"):
```

**Review by iker-barriocanal — COMMENTED** [review]
I'm not sure if I understand the approach with `SDK_CRASH_DETECTION_PROJECT_ID`, see my comment below.

File: src/sentry/tasks/post_process.py (hunk 1 above)
Comment (jjbayer) [reply src/sentry/tasks/post_process.py:1068]: @iker The `job["is_reprocessed"]` check is cheaper than the feature check, so I would keep this as-is.

File: src/sentry/tasks/post_process.py (hunk 2 above)
Comment (philipphofmann) [reply src/sentry/tasks/post_process.py]: Thanks 😃

File: tests/sentry/tasks/test_post_process.py [hunk 3]
```diff
@@ -1517,6 +1517,61 @@ def test_forecast_in_activity(self, mock_is_escalating):
         ).exists()
 
 
+class SDKCrashMonitoringTestMixin(BasePostProgressGroupMixin):
+    @with_feature("organizations:sdk-crash-detection")
+    @patch("sentry.utils.sdk_crashes.sdk_crash_detection.sdk_crash_detection")
```
[the remark this answers is not in the document]
Comment (philipphofmann) [reply tests/sentry/tasks/test_post_process.py:1522]: Good catch, thanks 🙏

File: src/sentry/conf/server.py [hunk 4]
```diff
@@ -3442,3 +3444,6 @@ class TopicDefinition(TypedDict):
 # The Redis cluster to use for monitoring the health of
 # Celery queues.
 SENTRY_QUEUE_MONITORING_REDIS_CLUSTER = "default"
+
+# The project ID for SDK Crash Monitoring to save the detected SDK crashed to.
+SDK_CRASH_DETECTION_PROJECT_ID: Optional[int] = None
```
[the remark this answers is not in the document]
Comment (iker-barriocanal) [reply src/sentry/conf/server.py]: Approving to unblock, but I still don't understand the approach. If the crashes of all organizations are going to be moved to a specific org-agnostic project, I think we should reconsider the approach.

**Review by iker-barriocanal — APPROVED** [review]

**Pull request merged**